### PR TITLE
wait for main window to start

### PR DESCRIPTION
**Pull request by iliabylich**
if we have multiple windows spectron waits for at least one of them to be running

### CODE REVIEW

File: lib/application.js [hunk 1]
```diff
@@ -43,11 +44,46 @@ Application.prototype.setupPromiseness = function () {
...
+      for (const windowId of windowIds) {
+        await self.client.window(windowId)
+        const url = await self.client.getUrl()
+        if (url.includes(self.mainWindowStringIdentifier)) {
+          cleanup()
+          resolve()
+          return
+        }
+      }
+      pollInterval = setTimeout(findMainWindow, 1000)
```
Comment (hurshagrawal) [inline lib/application.js]: `setInterval`?

File: lib/application.js [hunk 2]
```diff
@@ -43,11 +44,46 @@ Application.prototype.setupPromiseness = function () {
...
+    }
+
+    function cleanup () {
+      clearTimeout(timeout)
+      clearInterval(pollInterval)
+    }
+
+    findMainWindow()
+  })
+}
```
Comment (hurshagrawal) [inline lib/application.js]: Looks good but can we make some sort of helper that this uses so it's a bit easier to read? E.g.

```js
function pollWithTimeout(pollInterval, timeoutDuration, promise) { ... }
```

File: lib/application.js (hunk 1 above)
Comment (iliabylich) [reply lib/application.js]: No, this function is recursive

Comment (iliabylich) [reply lib/application.js]: I'll rename it to `nextTick`

Comment (hurshagrawal) [reply lib/application.js]: Then why `clearInterval` below instead of `clearTimeout`?

Comment (iliabylich) [reply lib/application.js]: good point 😄 Should be `clearTimeout`

**Pull request merged**